### DISCUSSION TITLE
Add IP with python custom script

### DISCUSSION THREAD

**Discussion by SRF-rueeggdo** [netbox-community/pynetbox#610] [Q&A] [answered]
Hello everyone

I am trying to create a custom script to automatically detect specific devices and assign them to an IP address but I'm running in some errors.
This is my sample code:

`    from ipam.models import IPAddress
import ipaddress

def checkAndSaveLoop(self, loopIps):
        ip = "10.120.23.193/26"
        ip = ipaddress.IPv4Network(ip)
        newLoopIp = IPAddress(
            address=ip,
            status=IPAddressStatusChoices.STATUS_ACTIVE,
        )
        newLoopIp.clean()
        newLoopIp.save()
`
        

I have tried several options:
1. pass IP as string (without mask)
Exception: AttributeError: 'str' object has no attribute 'prefixlen'
2. pass IP as string (with mask)
Exception: AttributeError: 'str' object has no attribute 'prefixlen'
3. pass IP as ipaddress.IPv4Network object (without mask)
Exception: AttributeError: 'IPv4Network' object has no attribute '_value'
4. pass IP as ipaddress.IPv4Network object (with mask)
Exception: ValueError: ip has host bits set

Does anyone have experience of how to create an IP for a custom script?

Netbox v3.6.4

**Comment by markkuleinio** [comment, marked as answer]
Since this has nothing to do with `pynetbox` package, you won't  maybe get much exact answers here. Better luck should be in the NetBox discussions: https://github.com/netbox-community/netbox/discussions

In any case, I don't think you need to use `ipaddress` module there at all, just get the syntax right otherwise.
